pyproject.toml

```diff
@@ -118,6 +118,13 @@ checks = ["all", #report on all checks, except the following
 exclude = ['__init__' ] # don't report on objects that match any of these regex
 
 [tool.mypy]
-follow_imports = 'skip'
-ignore_missing_imports = 'true'
-exclude = ["tests", "docs"]
+disable_error_code = ['import-not-found', # Unable to locate module or package specified
+                'import-untyped', # The import lacks type information
+                'no-untyped-call', # Function calls are only made to functions that are fully typed
+                'type-arg' ] # Requires type arguments to be specified in list[] and dict[]
+strict = true
+explicit_package_bases = true
+follow_imports = 'skip' #may want to remove
+exclude = ["tests"]
+packages = ["imap_processing" , "tools"]
+plugins = 'numpy.typing.mypy_plugin'
```

Comment:
That's great that you were able to define these here instead of `pre-commit-config`. I think its this team's preference to have all of these configurations under the one `pyproject.toml` umbrella!